### ISSUE TITLE
Simplify the Getting Started Page

### ISSUE THREAD

**Issue by mulyaj** [jamulussoftware/jamulussoftware.github.io#250]
**Is your feature request related to a problem? Please describe.**
Currently, I think the "Getting Started" page contains information irrelevant to _getting started_. For new or potential users, this page is probably the first page they click on after the home page. Thus, it addresses users that **have not yet** or **are about to** download and install the software. Therefore, sections like _Maximise Quality_ and _Have trouble? Can't keep in time?_  don't have relevance to the reader since they might not have even used the software yet.  

**Describe the solution you'd like**

The "Getting Started" page should ideally contain:

- Description of Jamulus 
- System + Hardware Requirements
- Recommended Hardware (Audio Interface and/or USB mic)
- Links to installation/set-up on Windows/Mac/Linux

 _Maximise Quality, Minimize Delay_ and _Having Trouble? Can't keep in Time?_ sections can be incorporated into the Onboarding page, describing an optimal first-time use of Jamulus.

In context, the flow for a new user would go from Getting Started (Check Hardware + Specs, determine if they can use Jamulus) -> Set-up on their OS (Follow installation steps and setup hardware) -> Onboarding (Learn to use Jamulus optimally)

**Comment by ann0see** [comment]
I just want to link the getting started page which will be part of the new version: https://github.com/jamulussoftware/jamuluswebsite/blob/changes/wiki/en/en-Getting-Started.md

> Description of Jamulus

In my opinion, this should be on the homepage https://jamulus.io

> System + Hardware Requirements

Have a look at the new page. Is this enough?

> Links to installation/set-up on Windows/Mac/Linux

This is already part of the sub-pages (Install for Windows/macOS/Linux)?

Sure, we can do some changes here.

**Comment by mulyaj** [comment]
I think I should clarify a little; I'm not advocating for adding anything new. The only changes would be that the content from _Maximise Quality_ and _Having Trouble _ be moved to onboarding.

To add, what I meant by "Description of Jamulus" was how it works (my bad).

**Comment by mulyaj** [comment]
Here's my [version of the page](https://github.com/mulyaj/jamuluswebsite/blob/experiment/wiki/en/en-Getting-Started.md) now. I've tried to streamline the information and added some formatting. Let me know your thoughts.

**Comment by ann0see** [comment]
Yes. This page looks more organized now. In the beginning, we had the "How Jamulus works" section right at the top but weren't sure if this might intimidate or confuse non-techy people. The recommended hardware section might be a bit misleading:

> Do I need an USB microphone and an interface?
 
Could be a question somebody might ask.

Are you sure we really want to move the "Minimize latency, maximize delay" section to the onboarding page?

**Comment by ann0see** [comment]
Furthermore, I'd like to make clear that you can use Jamulus without an external audio interface with ASIO4ALL and WiFi but that this will cause quality problems and therefore is not recommended. Otherwhise, some people might think "Ok. I can't use Jamulus since I don't own this special professional audio device thingy which is probably too expensive".

**Comment by mulyaj** [comment]
The "How Jamulus Works" section might need some reworking, but I do think there needs to be some explanation of what Jamulus does (I remember @Erioldoesdesign mentioning that the client-server aspect isn't apparent).

About the recommended hardware, the live site sort of already gives an imperative to the reader as it mentions. "Use an audio interface/external microphone, not your internal sound card", only after reading the body text do they realize that's not true. 

 I think there should be a delineation of the **minimum** (what is needed to run and use Jamulus) and **recommended** (what is needed to have an optimal experience with Jamulus). I guess for ASIO4ALL and Wi-Fi, they could be mentioned in the minimum requirements as **Audio Driver** (specifying OS etc.) and **Internet Connection**.

**Comment by ann0see** [comment]
> The "How Jamulus Works" section might need some reworking, but I do think there needs to be some explanation of what Jamulus does

Yes. There's also an issue called "diagramming Jamulus":  https://github.com/corrados/jamulus/issues/64

This might give some more insight? 

> "Use an audio interface/external microphone, not your internal sound card", only after reading the body text do they realize that's not true.

Yes. Probably it's too harsh a the moment. Since it's under "Maximize quality, minimize delay" we thought of this like a optional but strongly recommended section. But it seems as if this thought didn't come across.


> I think there should be a delineation of the minimum (what is needed to run and use Jamulus) and recommended (what is needed to have an optimal experience with Jamulus).

Yes. That's a good point! If we can clarify this that would be great.

**Comment by gene96817** [comment]
I recommend thinking of getting new users through two stages. The first, the minimal hardware configuration. Then if they wish, how to optimize performance. Consider the minimal configuration for getting on Jamulus as USB gamer headset (includes microphone) and USB ethernet adapter and cable. This is relatively low cost and the least intimidating. Then the user is ready for a discussion of choices to get more performance.

**Comment by ann0see** [comment]
>  The first, the minimal hardware configuration.

Yes. I agree for the website.

Nevertheless, we're currently trying the contrary approach (and are still waiting for the interfaces which we ordered last year, to arrive). I don't want that these people need to tinker around with ASIO4ALL. It just caused problems for me.

**Comment by mulyaj** [comment]
> There's also an issue called "diagramming Jamulus"

Thanks, I'll check this out.

> the minimal hardware configuration

I think we should be careful about the usage of the word "minimum". The documentation should be consistent in its usage and should only mean **the least required**. Thus, for hardware specifically, we could use "strongly recommended" for things like ethernet and wired headphones, since technically Jamulus could be run without them (albeit with a very poor experience).

**Comment by mulyaj** [comment]
Speaking of minimum requirements, what are the other requirements (memory, HD space, audio driver?).

**Comment by ann0see** [comment]
HD space: roughly about 70mb for Windows, but I'd suggest more if you want to use recording.

**Comment by gilgongo** [comment]
Just a note on the long-running issue of explaining how Jamulus works (and the related issue of the diagram):

Beyond the demographic of curious engineers, I'm not sure anyone really cares how Jamulus works. Vaguely comparable software like Zoom, Dropbox or WhatsApp don't feel the need to push explanations of their infrastructure, edge routing or PKI at their users because (despite these things being important parts of the system), nobody needs to know, nor does anyone ask.

But if they do, then I think it's reasonable to wonder why they would? What would they do, or how would they act differently if they knew that Jamulus was, for example, P2P and not CS, or that central servers don't carry the sound signals?

It is for these reasons that I think hitting new users in the face with a technical diagram is a mistake. And I say that as the person who created that diagram :-) It makes Jamulus look intimidating.

**Comment by trebmuh** [comment]
That makes sense.

**Comment by mulyaj** [comment]
> hitting new users in the face with a technical diagram is a mistake 

Yeah, I definitely agree with that. 

Would some kind of preview of Jamulus (screens of it in use) be better then? It would be beneficial to give presumptive users a peek into the interior. I know there's the wiki/demos page, but I feel like it's quite hidden at the moment.

**Comment by gilgongo** [comment]
Yes - in fact we probably do need to use more screenshots generally. There's [another ticket](https://github.com/jamulussoftware/jamuluswebsite/issues/192#issuecomment-746056697) for this in fact but it has a bit of dependency on sorting it out so that we can deal with images without going crazy

**Comment by gene96817** [comment]
We certainly don't need to explain how Jamulus works to be at the top level for all users. However, it should be easy to know where to get this information. This information is to enable the more technical user to understand the inputs and outputs through the UI for troubleshooting.  To be effective at troubleshooting, it is essential to know why. Currently, it is hard for me to know if the information I gather through word-of-mouth is well-grounded or not. (So here I am monitoring these threads.)

**Comment by gilgongo** [comment]
> To be effective at troubleshooting, it is essential to know why.

Apologies, but while I don't disagree with that idea in principle, I don't actually understand it in practice. 

Can you give some examples of problems you might have when using Jamulus that would be soluble through such knowledge? By this I mean something like "When I hear [something], I would know to adjust [system thing]". Or do you mean something more like, "I can't hear [something], so knowing about [system thing] will point me to doing [action]"?

I've tried to ask this question before but somehow never really get any clarity on it.

**Comment by gene96817** [comment]
@gilgongo  we are both having trouble getting clarity. For me, looking at the puzzle from the outside, I wish to have clarity on what can be fixed. Currently, I spend most of my puzzle-solving on, why is the overall delay so high. It would be great to be able to isolate the problem to a particular part of the data flow (as demarked by buffer boundaries or something else). Part of this question is are buffers lost or just late. The next problem I want to address is why is there a loss of fidelity. It would be great to be able to identify where in the signal path the problem is occurring. Finally, how do we do this when we are not at the client (meaning some less technical user is having trouble.

**Comment by gilgongo** [comment]
So if if latency is high (indeed for any given issue with Jamulus in fact), there are currently only a handful of things you can adjust to improve that: your jitter buffers, your sound card buffer, CPU or bandwidth hogs, switching servers or switching machines/hardware. In most cases, the only things you can practically do anything about are the jitter buffers and your sound card buffer settings (and maybe the small packets setting). 

The implication is that there might be some reliable connection between, say, a packet's lack of traversal of some part of the system and a remedial action like widening a sound card buffer. But is there such as connection? 

This is why I find hard to understand calls for more telemetry, flow diagrams etc. - how would more understanding change what you did?

**Comment by gene96817** [comment]
When the latency is high, I do want to know which of the handful of things need attention. I dislike the prospect of changing things experimentally to see if it gets better. Where I am, I see highly variable Internet performance. I would like to know if the problem is in the user's equipment or if it is the network. "Highly variable Internet performance" is a euphemism. It is hard to get network problems fixed without better data.

**Comment by gilgongo** [comment]
Ok so I think I'd be clearer if we reframe the issue as whether there is a reliable cause/effect pattern: something in the system that can reliably indicate a specific problem area (eg network issue Vs audio hardware) and therefore an effective action to take (eg use better Ethernet cable Vs adjust audio buffer) to solve that problem.

If there is such an indication, then we should pursue the surfacing of that. I don't know if that's possible though. It has been discussed a but I think though but never seems to get very far.

**Comment by gene96817** [comment]
That is the goal and it is hard. It is easier to have "test points" that can help a human isolate (or localize) the problem. There is a lot of awesome goodness in Jamulus. The technical (non-developer) users need tools to help Jamulus deliver. Most of the problems are caused by the users or the Internet. 

The model of what to do is to realize the user/developer intuitively knows what changes (adjustments) to make to optimize Jamulus operations. We need tools so that non-developers can do these tasks (which we call support).

**Comment by gilgongo** [comment]
From what I understand though, it's not possible to help a human isolate the problem, because there is no connection between a perceived problem like "popping audio" or "drop outs" and a programmatic event like a packet being dropped, queued or arriving late.

There are visual indications in Jamulus of programmatic events in the form of the delay and ping LEDs in the UI, but they're not actionable in the way you want because of my statements above. 

Does this make sense? Basically what I'm saying is that more "telemetry" can't help the problem of people not knowing which of the five things I've listed they need to tweak in the event of problems. 

Unless I'm wrong, you are, I'm afraid, doomed to blind experimentation in solving problems with Jamulus no matter how many logs, diagrams or flashing lights you have access to.

**Comment by gene96817** [comment]
You are mostly right.
I can't make a hard case for the telemetry. The telemetry provides the only clues we have for troubleshooting. In the end, telemetry is necessary because we cannot expect to always have an intelligent observer at the client with problems.

Dropped or late packets will always affect the audio codec. It is harder to connect because we might not hear the defect from a small (single packet?) defect. And a casual observer may not distinguish between different kinds of audio defects. This is hard with only human senses.

While you are mostly right, the only reason I have been willing to help so many users with their problems is that I am hoping to acquire enough experience with Jamulus's behavior to help us all escape the need for blind experimentation. Jamulus is awesome. It would be even more awesome if it was easy to install and easy to support.

**Comment by gilgongo** [comment]
When you say telemetry provides the only clues we have for troubleshooting, I disagree. Your ears are the only real clue. Everything else is (potentially) misleading. 

Be that as it may, you say you are hoping to acquire enough experience to help us all escape the need for blind experimentation. Do you think you've seen any cause/effect patterns (and crucially) actions that could be documented for the purposes of troubleshooting?

**Comment by gene96817** [comment]
Thank you for your persistence.
The ears give the first indication that the whole system is not performing as expected.
From the input codec to the output codec, late or missing packets introduce sound defects. If the codecs are connected by a wire, the only audio errors are from bit errors. If you add the Internet, then we add packet delay and jitter which causes late or lost packets. We add buffering (and delay) to compensate for packet delay/jitter. Hence, it would be good to know at what stage of the digital datapath are packets late or lost.

If the packets were numbered, we could detect lost packets and we can measure if packets interarrival intervals to detect which stages of processing contribute to packets being (too) late. Because we don't have this information, I am scrutinizing my nmon reports looking for clues. The nmon data are only very indirect clues.

With the digital path understood, the other source of audio defects would be on the analog side of the codecs. Then we are talking about microphone sensitivity and frequency response, preamp operation (noise and distortion), and headphone frequency response.

I hope this is useful.

**Comment by gilgongo** [comment]
So imagine you had all that information. What would you do in Jamulus other than prod some buffers and check if someone isn't watching Netflix? Isn't that the same as what you are doing now?

**Comment by gene96817** [comment]
No, If I knew why, where the problem was occurring, I would not be spending time chasing shadows. I could know if the network link had enough capacity for both Netflix and Jamulus. Currently, I spend a lot of time trying to get a user onboard hoping things will work well enough to keep the ensemble together. At present, we can't know that a Netflix watcher is to blame. That is an easy scapegoat.

**Comment by gilgongo** [comment]
>  If I knew why, where the problem was occurring

My intuition is that the "why" there (beyond the information Jamulus already shows) is unknowable as I have explained. Perhaps somebody more knowledgeable than me can confirm that.

**Comment by gene96817** [comment]
To detect these conditions, first, the packet (payloads) send over UDP needs to be numbered. This will enable the detection of lost packets. Then it is necessary to track the packet interarrival rate. That will enable knowing when a packet is excessively late. Then the problem becomes transparent.  

As an added bonus, when an event causes one stream to have a significant delay, all the late and useless packets can be flushed and the system will recover faster.  Without this flushing, one user's traffic could have an offset that is persistent, and that user's sound could sound late until an event resets all the sound streams (a resynchronization event). (I believe we are observing this present offset behavior when one of our sessions has a visitor from 4 timezones away.)

**Comment by ann0see** [comment]
To come back to the topic of this issue: 

> Maximise Quality, Minimize Delay and Having Trouble? Can't keep in Time? sections can be incorporated into the Onboarding page, describing an optimal first-time use of Jamulus.

Any more opinions on this?
How visible/invisible should this be?

@mulyaj

**Comment by pljones** [comment]
Also...
>  If a server is made public and registered on a central server, its information will be broadcasted to all clients.

This isn't clear -- it sounds as if making a server public on a central server broadcast audio to all clients -- whether connected to _your_ server or somewhere else!  Which, of course, it doesn't.  In fact, making a server public has no effect on where the audio goes.

More accurate:
>  If a server is made public - that is, registered on server list - anyone accessing that server list will see it listed and be able to connect to it.

(NOT on this page, somewhere we need to explain about official server lists, custom/private server lists and "private" servers.)

**Comment by gilgongo** [comment]
To summarise: 

We make the assumption that there are two distinct "phases" of starting out with Jamulus (downloading, installing and setting up audio hardware to work with it, then connecting to a server and playing with people for the first time).

If so, and because Getting Started existed before we wrote Onboarding, then we could make a decision to split some things out from one to the other. 

In doing this, we also assume that people will dutifully read **both** Getting Started and Onboarding. When I wrote the former, I assumed few would read much before they clicked the download link, so I put a few minimal "onboarding" things  on that page (maximise quality, etc,) in the hope they'd skim those and bear them in mind when they got to the point of connecting.

But if we do think that people _will_ read everything, then let's split things out so they need to read both sections. The risk here is that we see an uptick in people grabbing Jamulus, firing it up and then not knowing it's going to suck on wi-fi or something.

**Comment by gene96817** [comment]
I agree that there is a "significant" risk that some people will just skip the documentation. If and when they fail, it should be possible for them to cleanly start over and read the information we provide. I expect people will be more interested in reading the information after they have failed their first try.

**Comment by pljones** [comment]
I think the primary things to prevent a first time user doing that could cause them simply to bin Jamulus are:
* using their laptop mic and speakers - e.g. ASIO4ALL with all inputs and outputs enabled; can also happen on MacOS, I think
* not as immediate, using WiFi

Different people will have different experiences with their WiFi.  However everyone trying out ASIO4ALL needs to know how to set it up right.

Both probably affect the more naive users but neither is likely to make them think "Oh, I better read the manual" -- it's more likely to make them think "Oh, this just doesn't work".  So it's best to prevent them before they start...

**Comment by dcorson-ticino-com** [comment]
I think we need to start by lowering any hesitation and showing that Jamulus is easy-> i.e. minimum needs, showing pictures of what users should expect (as is done in onboarding) and then mentioning how to upgrade.
I would also turn the "onboarding" page into "Getting Started" or "First Steps" with the program installation now in getting started as a link to "Program Installation" or such as that is really not the problem.  
BTW "Onboarding" is horrible "Human Resources" talk and not fit for use among normal human beings IMHO (immediate association waterboarding).
I am attaching my "Setting up Jamulus in Five Steps" guide that I have been using successfully.  It is targeted just to the orchestra I play with, but I think shows how easy it is to start out using Jamulus.
[JamulusQuideEn.pdf](https://github.com/jamulussoftware/jamuluswebsite/files/5896563/JamulusQuideEn.pdf)

**Comment by gilgongo** [comment]
I would say that there's certainly a case for combining Getting Started and Onboarding, if only because the former was written first and may not sit well with the latter now perhaps.

However, @dcorson-ticino-com I think there are probably two competing approaches to the problem of getting people started with software in general: 

1. Tell them what they need to do comprehensively, but as briefly as possible (I would say this is the approach in your "Five Steps"). If they read and follow the guide, the chances of subsequent problems will be minimal.
2. Assume they won't read much more than a few lines before they double-click on the installer in anticipation of everything being easy. Provide a guide for them to use if - and they may not - have problems. This is the approach that most large or commercial projects take (Zoom, or Plex for example whose "Getting Started" are pretty much just download buttons).

The rationale behind 2 is that there is also a phenomenon whereby people load a page full of (to them) very detailed instructions and think _"Oh OK. This looks too difficult. Back to Google then ..."_ 

So far, we have been in the latter camp: brief "need to know" (Getting Started), then helping hand in the event of issues (Onboarding). But ultimately, this is a sort of "cultural" thing. Do we think our audience responds best to approach 1 or 2? Approach 1 is increasingly unusual.

EDIT: It occurs to me that the overall "brand" of Jamulus is possibly one that lends itself to the first approach, which in turn attracts those who respond to approach 1, and perhaps repels those who don't? So far, this hasn't really been a problem, but as of the last couple of months, we are staring to appear in Google searches (following @ann0see work to get us off Github wiki and other search engine friendly moves). So that may change, 

ANOTHER edit (this is getting too long, sorry): Don't forget also that inducting choir members is a fundamentally different scenario to that of somebody coming to Jamulus for the first time. In the former case, if they can't use Jamulus, they're not going to be in the choir. In the latter, there is always JamKazam, NinJam, Jamtabaa, etc.

**Comment by dcorson-ticino-com** [comment]
Good thoughts. 
I think the basic difference is that sites like Zoom are pure PR events.  There is no actionable information, just pretty pictures and the promise of a perfect world, you just need to click.  If you need help there is nothing (but google).

While we, of course, want to convince people to use Jamulus, we are not selling anything, we are offering good product and a helping hand to get started as easily as possible.  I see there two very different worlds.  I think we should not underestimate the reaction of people to a helping hand.

**Comment by gilgongo** [comment]
I don't really agree that the commercial considerations of those systems are that relevant (they do in fact have very extensive documentation, should people wish to click on "help"). VLC is free. [Tor](https://www.torproject.org/) is free. They are (arguably) rather more complex in some ways than Jamulus. In fact Tor a very good example of what I mean.

> I think we should not underestimate the reaction of people to a helping hand.

I agree, but my point is a "cultural" (perhaps even ideological) one. It starts from a fear that Jamulus will forever present itself as difficult. An engineer's tool. For people with high computing skills or knowledge of things like audio interfaces, mixing, "client" and "servers". For every well-meaning "helping hand" there is a possibility that it aggregates to slap in the face by mistake.  

Of course, we might legitimately say "Well, Jamulus **is** for those people - it's **not** easy to use, and needs to be explained." After all, our "shop window" as it were, isn't as sexy as something like Tor or Zoom. And that's a legitimate point. It's probably true for now. But I think we should not underestimate the reaction of people to that approach in the longer term.

**Comment by gene96817** [comment]
We do have a cultural bias that makes Jamulus difficult for non-engineering musicians. A lot of our solutions require being a server administrator. That is a big barrier for many people. Of course, there are a lot of people where administering a server is a basic skill.

**Comment by gilgongo** [comment]
Well, server setup and maintenance is a specialist skill (beyond simply spinning up a small public server to see if that works with some friends - that's trivial). 

Fundamentally, you don't need to run a server to use Jamulus and when I wrote the server docs I was almost _trying_ to make them sound complicated for that reason. But this is a [totally separate conversation](https://github.com/jamulussoftware/jamulus/issues/748#issuecomment-770191090).

**Comment by gene96817** [comment]
"Well, server setup and maintenance is a specialist skill (beyond simply spinning up a small public server to see if that works with some friends - that's trivial)."

Apparently, I have misunderstood the frequent discussions about spinning up a (small) public server for demo/test purposes. It was never clear to me that comments on how easy it is to just run a server were not a recommended (and routine) practice.

**Comment by mulyaj** [comment]
>Assume they won't read much more than a few lines before they double-click on the installer in anticipation of everything being easy. Provide a guide for them to use if - and they may not - have problems. This is the approach that most large or commercial projects take (Zoom, or Plex for example whose "Getting Started" are pretty much just download buttons).

I definitely agree with this approach.

Currently, I think the content in the Getting Started page sits in a weird spot where different types of information with different values are presented to the user. For us, we know that this is pertinent information and we believe we must tell them as soon as possible, therefore we think the solution is easy: just put it at the beginning!

However, new users, especially those that may come through a google search, will have significantly less time they want to invest. These new users are simply exploring and are in a discovery phase. When we start telling them at the beginning, "you need to **do this**, that and **also this**! but **remember don't do this!**", new users simply won't read it, or at most, skim the heading and forget it.

If we want to provide proactive help (in the case of onboarding and setup), I believe that the best place to do that would be directly in the software itself. In my opinion, the website should allow users to get started quickly and also be the place where we can provide reactive help (FAQ, Troubleshooting, Documentation).

At the moment, I understand that large UI changes in the software aren't entirely actionable (correct me if I'm wrong). That's why I'm advocating for moving certain sections of the Getting Started page to the Onboarding page. That way, we can better follow along with a user's task flow, while guiding them in the right direction.

**Comment by mulyaj** [comment]
I do agree with @dcorson-ticino-com that "Onboarding" might be too HR. Maybe something like "Using Jamulus for the first time" would be better?

**Comment by pljones** [comment]
Random thoughts...

Try this as a first time user.
1. Go to jamulus.io
2. Read the banner ... where's download?  "Advanced: download now"

Why is it "Advanced" to "download now"?  That immediately says: "TL/DR - go away"

That "Get started now" should be "Download now", to an OS-detecting link that kicks off the latest download immediately.  Then underneath it should be the "Get started with Jamulus" link or something like that.

I'd also have the layout of the banner more horizontal, with the cloud to one side to the left of the rest, with the "Jamulus | Play music online. With friends.  For free." a bit larger, taking up about the same space as the (now) "Download" and "Getting start" (whatever) link.  The page is less likely to scroll then...

I also think @dcorson-ticino-com's document is pretty good, but would indeed need a bit of work to make it appeal to the stand alone user.  However, neither the current "Getting started" nor the "Onboarding" (name must change...) pages really get a user started.

The best would be separate paths for each OS.  Windows needing the most hand-holding, of course...

Getting started should:
- get the user started quickly (again, big "Download" button that kicks of the download on their platform in case they missed it)
- avoid them having problems (like not having turned off direct monitoring, not having ASIO on Windows, not having 48K on all platforms, any other minimal requirements - not too much detail; "if you can hear your sound when not connected, fix it here" link)
- walk them through getting connected - this needs to explain how to see when a server is full, to avoid disappointment and what the ping times mean, again to avoid disappointment (AND KEEP YOUR MOUSE OFF THE LIST until it settles :) )
- tell them where more help is available

Getting started should not
- explain what Jamulus is (front page should do that or do it elsewhere linked from front page)
- explain how Jamulus works (front page should _not_ do that, but it could link to it)
- not say "Summer 2020! Tell us about Jamulus" - it's Winter 2021 now, at least in the northern hemisphere.
- be a trouble shooting guide, not even "common problems" (too off-putting - should be linked, though)

**Comment by gene96817** [comment]
I like the direction. Regarding "appeal to the stand alone user", many musicians are being introduced (pushed?) into Jamulus because their ensemble (chorus) will use Jamulus. They need to feel reassured when they first visit jamulus.io.

**Comment by pljones** [comment]
I'm being off topic still..

Another front page point should be to feature "What are people doing with Jamulus?", and encourage projects to get in touch.

**Comment by gilgongo** [comment]
@pljones Yes, I think this highlights the fact that the "journey" has been a bit too fragmented in its creation and your suggested modifications are good ones - if only because I have suggested some of them in the past :-). 

BTW as I've just said in [another discussion](https://github.com/orgs/jamulussoftware/teams/designers/discussions/4) (there are a lot of them): do we think it might be an idea (along with your OS-specific observation) of customising the new user journey by intended outcome? So if we established for example if they are a singer (mic only), or a singer/player (instrument + mic), and then acoustic instrument vs electric? Would that mean we could bypass certain problems or emphasise certain issues in the setup? For example, Rule Number One is of course far harder to follow if you are singing or playing an acoustic instrument.

@gene96817 

> They need to feel reassured when they first visit jamulus.io

Just so I'm clear - do you mean reassured by seeing information rather than just a download link? In other words the first approach I outlined in https://github.com/jamulussoftware/jamuluswebsite/issues/250#issuecomment-770177375? I think this discussion is solidifying around which "direction" we want to take on this.

Re. "onboarding" feeling bad - we might be able to side-step the issue if we re-cast "Getting Started" as one page as per @pljones  description of "Getting started should" here https://github.com/jamulussoftware/jamuluswebsite/issues/250#issuecomment-770302625

**Comment by gene96817** [comment]
I like the direction of this issue.

 When I say "reassured", I am thinking of the technically nervous musician. I just want to keep a focus on reassuring those musicians that they don't need to be a programmer or tech wiz to succeed in using Jamulus.

**Comment by gilgongo** [comment]
> reassuring those musicians that they don't need to be a programmer or tech wiz to succeed in using Jamulus.

Yes, although that's easier said than done because of the subtle issues around things like the natural tendency to want to "explain" in various ways (the diagram conversations just never end), the current limitations of Jamulus (eg it may not be possible to make it any easier to choose and configure the right audio interfaces), and "soft" issues around website design and presentation (limited resources to make a very comforting-looking website, and design efforts coloured by our desire to keep things collaborative as an open source project).

But I think that we are getting there, and @pljones suggestions seem to me to be the most lucid so far.

**Comment by Hk1020** [comment]
> I think we need to start by lowering any hesitation and showing that Jamulus is easy-> i.e. minimum needs, showing pictures of what users should expect (as is done in onboarding) and then mentioning how to upgrade.
> I would also turn the "onboarding" page into "Getting Started" or "First Steps" with the program installation now in getting started as a link to "Program Installation" or such as that is really not the problem.
> BTW "Onboarding" is horrible "Human Resources" talk and not fit for use among normal human beings IMHO (immediate association waterboarding).
> I am attaching my "Setting up Jamulus in Five Steps" guide that I have been using successfully. It is targeted just to the orchestra I play with, but I think shows how easy it is to start out using Jamulus.
> [JamulusQuideEn.pdf](https://github.com/jamulussoftware/jamuluswebsite/files/5896563/JamulusQuideEn.pdf)

A comment to your guide. Add a point to ask people if they can hear any sound from YouTube or the like. Or if they can use zoom. First make sure the equipment is working. (Close Jamulus though, they don’t work at the same time). In fact, if they can use zoom (many people do in these times) they can use Jamulus. And more than once I told people to raise the normal Windows audio level when they couldn’t hear us in Jamulus. Didn’t occur to them. 

And there is the issue with the unsigned installers. People see scary messages of dangerous downloads and so on. I know why but they need to be encouraged to try anyway.

**Comment by Hk1020** [comment]
> However, @dcorson-ticino-com I think there are probably two competing approaches to the problem of getting people started with software in general:
> 
> 1. Tell them what they need to do comprehensively, but as briefly as possible (I would say this is the approach in your "Five Steps"). If they read and follow the guide, the chances of subsequent problems will be minimal.
> 2. Assume they won't read much more than a few lines before they double-click on the installer in anticipation of everything being easy. Provide a guide for them to use if - and they may not - have problems. This is the approach that most large or commercial projects take (Zoom, or Plex for example whose "Getting Started" are pretty much just download buttons).
> 
> The rationale behind 2 is that there is also a phenomenon whereby people load a page full of (to them) very detailed instructions and think _"Oh OK. This looks too difficult. Back to Google then ..."_

Absolutely. No 1 is out these days. People don’t read anything. I am not much different. Stuff just has to work (and Jamulus does!). 

...
> ANOTHER edit (this is getting too long, sorry): Don't forget also that inducting choir members is a fundamentally different scenario to that of somebody coming to Jamulus for the first time. In the former case, if they can't use Jamulus, they're not going to be in the choir. In the latter, there is always JamKazam, NinJam, Jamtabaa, etc.

Correct. I am in the choir camp and there most people are afraid of computers.

**Comment by gilgongo** [comment]
@Hk1020 OK so I think we are gaining agreement around PLjones's prescription in https://github.com/jamulussoftware/jamuluswebsite/issues/250#issuecomment-770302625

BTW one thing that hasn't been raised this time around is the issue of "beginners vs power users". It has been suggested in the past that we divide the docs along those lines, but I am generally opposed to that approach as I think in practice it just gets confusing and difficult for various reasons.

**Comment by pljones** [comment]
There are "first time users".  I think we're all agreed that, whether they're "power users" or not, they want a very easy way to get the software actually running so they can see what it's all about.  That needs to be made as simple as possible for everyone.

Just to note on that -- it also needs to be made as simple as possible for people who are "pure MIDI", i.e. no audio in to their computer.  I'd break that out onto a separate page though, early on in the process.  Those users are probably already more advanced and able to handle more complex problems.  But still catch them early so they feel welcome and supported.

Beyond that, many users won't want any more - so long as when they play or sing, they're hearing and getting heard, they're done.

Anyone who does come looking for more could probably be called a power user of sorts.  But not all power users are technical.  Most just want solutions to more complex problems.  So there's the "how to guides".   What questions are most often asked?  "How do I set up with Zoom?" etc?  Each of those could end with a "how it works" section so the real power users can understand it and adapt it.

Oh - not to stay on topic too long... - and there should also be a more extensive "How to get involved" section, especially with the re-organisation.  It currently feels like you need to be a developer.  It could have sections on:
* "How to raise a new idea" (check it's not already done, check it's not already raised, check if it's likely to get accepted - read the guidance notes, etc)
* "How to report a bug" (check you're on the latest version, check the server is on the latest version, write down the steps to reproduce the bug, say what you expected to happen, say what actually happened)
those two shouldn't just be on the templates: the guidance needs to happen first, the templates are reminders
* "What to do if you have a code change to submit" (check if it's likely to get accepted - read the guidance notes, make sure it's passing the automated checks).

**Comment by dcorson-ticino-com** [comment]
> "What to do if you have a code change to submit" (check if it's likely to get accepted - read the guidance notes, make sure it's passing the automated checks).

Hey!  I need this quick, I didn't even know there are automated checks !
I am simply using QT Creator, is that the right compiler ? build ?
I have found no info what I should be doing, but maybe I am blind.  
Please point me in the right direction.

**Comment by gilgongo** [comment]
@dcorson-ticino-com @pljones 

The word "Contribute" is on the home page (and the footer of each page after that). It has a link to [this page](https://jamulus.io/wiki/Contribution), which I guess could be improved (in fact I think we have a version of it on a branch somewhere?)

These is also [this page](https://jamulus.io/wiki/Compiling), but I can't remember where it's linked from (which reminds me, I think I may just go in and manually restore all the breadcrumbs we lost when we moved from Github wiki). Maybe it got forgotten?

If somebody wants to write a Jamulus code submission HOWTO with stuff like what compilers and builds to use _and keep that up to date_, then that might be a good idea. I don't know as I'm not an engineer.  

BTW I'm a little circumspect about promoting efforts to invite contribution since it may imply we can digest all submissions, triage and de-dupe stuff properly. Which we can't really. But maybe that's just the nature of things.

**Comment by pljones** [comment]
> The word "Contribute" is on the home page (and the footer of each page after that). It has a link to [this page](https://jamulus.io/wiki/Contribution), which I guess could be improved (in fact I think we have a version of it on a branch somewhere?)

"Contribute" can means "give us money".  It's not a terribly inviting word and, as a single word, a bit stark.  And yes, it was the existing page I was suggesting could do with an overhaul, but I'm freely admitting that's out of scope here..! :)

**Comment by pljones** [comment]
> BTW I'm a little circumspect about promoting efforts to invite contribution since it may imply we can digest all submissions, triage and de-dupe stuff properly. Which we can't really. But maybe that's just the nature of things.

A separate issue, really - leadership need not be about _doing_, it can be just about _guiding_ and helping those who do, do well.

I think the spirit of openness needs to be pervasive.  There should be no feeling that people are excluded from being involved.  Perhaps the best some could do would be to triage issues - going through reported bugs to ensure they're properly documented and real, etc, clarifying requests for new features and so on.

**Comment by gilgongo** [comment]
> "Contribute" can means "give us money". 

That's certainly one meaning of the word. Unfortunately, all other synonyms in English have the same connotation (help out, donate, chip in...). It's sufficiently contextualized on the homepage to avoid confusion. Otherwise, I'm at a loss to come up with anything that means non-fiscal contribution of resources :-)

**Comment by gilgongo** [comment]
> need not be about doing, it can be just about guiding

Indeed, I make no distinction.

As to pervasion of openness, I assume (but I could be wrong) that if people know anything about FOSS then the will be in no doubt about that.

The subject of management in general is the subject of an upcoming f2f discussion though. This sounds like one for the agenda (once date and time are agreed).

**Comment by gene96817** [comment]
catching up with the discussion from @Hk1020 (5 hours ago at this writing). There is another way to merge 1 and 2. In both cases, the user goes straight to installation, and then there needs to be a clear problem resolution path for any issues that arise. In this path, Jamulus must self-diagnose the problem and tell the user what needs to be done to resolve the problem.

My plan B is to take the user through a step by step preinstallation check to ensure Jamulus does encounter a problem and then a step by step configuration of Jamulus. Clearly, plan B does not work if the user will not read and follow all the steps,.

Helping users with their install problems is no fun and getting very old.
